pennylane/gradients/parameter_shift_cv.py

```diff
@@ -425,7 +425,7 @@ def processing_fn(results):
         start = 0
 
         if not results:
-            results = [np.zeros([tape.output_dim])]
+            results = [np.squeeze(np.zeros([tape.output_dim]))]

```

Comment:
How come this needs to be squeezed? 🤔 

Would this be solved by just having
```python
results = [np.zeros(tape.output_dim)]
```
(e.g., removing the inner square brackets)

Reply:
I tried it without brackets but `np.zeros(1)` will produce a 1D array, so it needs to be squeezed. The only way to make it a scalar is via `np.zeros(())` I think, but that would require additional logic anyways.

Reply:
Regarding the fix in this file, it's just that the post-processing function attempts to stack the result of a tape execution:
https://github.com/PennyLaneAI/pennylane/blob/1c0a57c68e8e7d237bbad29109053dd70e9d3716/pennylane/gradients/parameter_shift_cv.py#L445-L449
This fails when the tape result is a scalar, so I just give the tape result at least 1 dimension and undo it afterwards if it was a scalar.
https://github.com/PennyLaneAI/pennylane/blob/03479ac6516680d36edf232653dc3e44e6eeecb7/pennylane/gradients/parameter_shift_cv.py#L450-L451
https://github.com/PennyLaneAI/pennylane/blob/03479ac6516680d36edf232653dc3e44e6eeecb7/pennylane/gradients/parameter_shift_cv.py#L458

I'm not sure why the stacking is there, is the execution result of a single tape not already a tensor, or are there instances where that is not the case?

Reply:
> Regarding the fix in this file, it's just that the post-processing function attempts to stack the result of a tape execution:

Most likely, it was there because I discovered originally that adding the stack caused the tests to pass, and I didn't question it enough 😬